src/backend/distributed/utils/statistics_collection.c

```diff
@@ -0,0 +1,155 @@
+/*-------------------------------------------------------------------------
+ *
+ * statistics_collection.c
+ *	  Anonymous reports and statistics collection.
+ *
+ * Copyright (c) 2017, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#if HAVE_LIBCURL
+
+#include "postgres.h"
+
+#include <curl/curl.h>
+#include <sys/utsname.h>
+
+#include "access/xact.h"
+#include "citus_version.h"
+#include "distributed/metadata_cache.h"
+#include "distributed/statistics_collection.h"
+#include "distributed/worker_manager.h"
+#include "lib/stringinfo.h"
+
+bool EnableStatisticsCollection = true; /* send basic usage statistics to Citus */
+
+static uint64_t NextPow2(uint64_t n);
+static uint64_t ClusterSize(List *distributedTableList);
+static bool SendHttpPostRequest(const char *url, const char *postFields);
+
+/*
+ * CollectBasicUsageStatistics sends basic usage statistics to Citus servers.
+ * This includes Citus version, table count rounded to next power of 2, cluster
+ * size rounded to next power of 2, worker node count, and uname data.
+ */
+void
+CollectBasicUsageStatistics(void)
+{
+	List *distributedTables = NIL;
+	uint64_t roundedDistTableCount = 0;
+	uint64_t roundedClusterSize = 0;
+	uint32_t workerNodeCount = 0;
+	struct utsname unameData;
+	StringInfo fields = makeStringInfo();
+
+	StartTransactionCommand();
+	distributedTables = DistributedTableList();
+	roundedDistTableCount = NextPow2(list_length(distributedTables));
+	roundedClusterSize = NextPow2(ClusterSize(distributedTables));
+	workerNodeCount = ActivePrimaryNodeCount();
+	CommitTransactionCommand();
+
+	uname(&unameData);
+
+	appendStringInfo(fields, "citus_version=%s", CITUS_VERSION);
+	appendStringInfo(fields, "&table_count=" UINT64_FORMAT, roundedDistTableCount);
+	appendStringInfo(fields, "&cluster_size=" UINT64_FORMAT, roundedClusterSize);
+	appendStringInfo(fields, "&worker_node_count=%u", workerNodeCount);
+	appendStringInfo(fields, "&os_name=%s&os_release=%s&hwid=%s",
+					 unameData.sysname, unameData.release, unameData.machine);
+
+	SendHttpPostRequest("http://localhost:5000/collect_stats", fields->data);
```

Comment:
I'll replace this with a public https URL.

Reply:
Is there a timeout on this?

Reply:
@marcocitus: just added a timeout of 5s.